src/graph_export/rerank_graph_cli_parser.cpp

```diff
@@ -0,0 +1,137 @@
+// limitations under the License.
+//*****************************************************************************
+#include "rerank_graph_cli_parser.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <optional>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include "../capi_frontend/server_settings.hpp"
+#include "../ovms_exit_codes.hpp"
+#include "../version.hpp"
+#include "../status.hpp"
+
+namespace ovms {
+
+RerankGraphSettingsImpl& RerankGraphCLIParser::defaultGraphSettings() {
+    static RerankGraphSettingsImpl instance;
+    return instance;
+}
+
+void RerankGraphCLIParser::createOptions() {
+    this->options = std::make_unique<cxxopts::Options>("ovms --pull [PULL OPTIONS ... ]", "-pull --task rerank graph options");
+
+    // clang-format off
+    options->add_options("rerank")
+        ("graph_target_device",
+            "CPU, GPU, NPU or HETERO, default is CPU.",
+            cxxopts::value<std::string>()->default_value("CPU"),
+            "GRAPH_TARGET_DEVICE")
+        ("num_streams",
+            "The number of parallel execution streams to use for the model. Use at least 2 on 2 socket CPU systems.",
+            cxxopts::value<uint32_t>()->default_value("1"),
+            "NUM_STREAMS")
+        ("max_doc_length",
+            "Maximum length of input documents in tokens.",
+            cxxopts::value<uint32_t>()->default_value("16000"),
+            "MAX_DOC_LENGTH")
+        ("version",
+            "Version of the model.",
+            cxxopts::value<uint32_t>()->default_value("1"),
+            "VERSION");
+}
+
+void RerankGraphCLIParser::printHelp() {
+    if (!this->options) {
+        this->createOptions();
+    }
+    std::cout << options->help({"rerank"}) << std::endl;
+}
+
+void RerankGraphCLIParser::parse(const std::vector<std::string>& unmatchedOptions) {
+    try {
+        if (!this->options) {
+            this->createOptions();
+        }
+        std::vector<const char*> cStrArray;
+        cStrArray.reserve(unmatchedOptions.size() + 1);
+        cStrArray.push_back("ovms graph");
+        std::transform(unmatchedOptions.begin(), unmatchedOptions.end(), std::back_inserter(cStrArray), [](const std::string& str) { return str.c_str(); });
+        const char* const* args = cStrArray.data();
+        result = std::make_unique<cxxopts::ParseResult>(options->parse(cStrArray.size(), args));
+
+        if (result->unmatched().size()) {
+            std::cerr << "error parsing options - unmatched arguments: ";
+            for (auto& argument : result->unmatched()) {
+                std::cerr << argument << ", ";
+            }
+            std::cerr << std::endl;
+            exit(OVMS_EX_USAGE);
+        }
+    } catch (const std::exception& e) {
+        std::cerr << "error parsing options: " << e.what() << std::endl;
+        exit(OVMS_EX_USAGE);
+    }
+}
+
+void RerankGraphCLIParser::prepare(ServerSettingsImpl* serverSettings, ModelsSettingsImpl* modelsSettings) {
```

Comment:
W only use hfSettings here so why use modelsSettings & serverSettigns?

Reply:
if (!this->validate(serverSettings)), the validation may need to check for server settings. Model settings used to get the model name.